tests/gassubsidies/gas_subsidies_test.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2025 Sonic Operations Ltd
+// This file is part of the Sonic Client
+//
+// Sonic is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Sonic is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with Sonic. If not, see <http://www.gnu.org/licenses/>.
+
+package gassubsidies
+
+import (
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/0xsoniclabs/sonic/tests"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGasSubsidies_CanBeEnabledAndDisabled(
+	t *testing.T,
+) {
+	require := require.New(t)
+
+	// The network is initially started using the distributed protocol.
+	net := tests.StartIntegrationTestNet(t)
+	upgrades := map[string]opera.Upgrades{
+		"sonic":   opera.GetSonicUpgrades(),
+		"allegro": opera.GetAllegroUpgrades(),
+		//"brio":  opera.GetBrioUpgrades(),
+	}
+	for name, upgrade := range upgrades {
+		t.Run(name, func(t *testing.T) {
+			client, err := net.GetClient()
+			require.NoError(err)
+			defer client.Close()
+
+			// enforce the current upgrade
+			tests.UpdateNetworkRules(t, net, upgrade)
```

Comment:
This might or might not work depending on the execution order of the tests. An upgrade from Sonic to Allegro is fine, while a "downgrade" from Allegro to Sonic is denied.

Note that the iteration order through maps is always random in Go.